### DISCUSSION TITLE
getServerSideProps() doesn't have access to cookies on first render after redirect but available on refresh

### DISCUSSION THREAD

**Discussion by slowfound** [vercel/next.js#39613] [Help]
I'm serving cookies from my server-side api:

```javascript
res.setHeader('Set-Cookie', AJWT)
res.redirect(302, '/')
```

in my `index.js` file I'm trying to get the cookie before the page is rendered:

```javascript
export async function getServerSideProps(context) {
    let cookies = context.req.headers.cookie

    if (typeof cookies !== 'string') {
        return {
            props: { auth: false },
        }
    } else {
        const { AJWT } = cookie.parse(cookies)

        return {
            props: { auth: AJWT ? true : false },
        }
    }
}
```

on the first render after the redirect, cookies are not available but when I refresh they are successfully caught. Also worth mentioning that the cookie is available on my devtools when I'm first redirected.

Is there a way to make the cookies available on the first render?

**Comment by guigonzalezz** [comment]
Does someone already solved this problem?

**Reply by slowfound** [reply]
It ended up being an issue with the `SameSite` property on the cookie. It should be set to `lax` or `none` for it to be accessible on the first render.
